### PR TITLE
remove SpaceMapRBTree

### LINKED ISSUE #6289: SpaceMapRBTree is not used anymore
## Enhancement

We better remove it. Because it is not maintained anymore.

### PR DESCRIPTION
**Pull request by flowbehappy**
### What problem does this PR solve?

Issue Number: close #6289

Problem Summary:

### What is changed and how it works?

Remove SpaceMapRBTree

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/spacemap/SpaceMap.h [hunk 1]
```diff
@@ -135,8 +134,6 @@ class SpaceMap
     {
         switch (type)
         {
-        case SMAP64_RBTREE:
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/spacemap/SpaceMap.h:138]: Also remove these files `dbms/src/Storages/Page/V3/spacemap/RBTree.{h,cpp}`

**Review by lidezhu — APPROVED** [review]

File: dbms/src/Storages/Page/V3/spacemap/SpaceMap.h (hunk 1 above)
Comment (flowbehappy) [reply dbms/src/Storages/Page/V3/spacemap/SpaceMap.h:138]: ok

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**